### DISCUSSION TITLE
Custom tailwind class pattren matcher for Leptos

### DISCUSSION THREAD

**Discussion by s1n7ax** [tailwindlabs/tailwindcss#13684] [Help] [answered]
In rust Leptos framework, attributes can be defined as follows. Syntax is `<attribute>:<class_name>=<boolean>`. If the boolean is true, class name will be added to the existing list. Here, tailwind cli by default identify the `class="ease-in-out transition-all duration-1000"` attribute as css classes and successfully generates it to the output css file. However, it's not capturing any of `class:<class_name>` pattern. Is there a way to add custom match patterns in tailwind config? 

```rust
<h1
  class="ease-in-out transition-all duration-1000"
  class:translate-x-20=move|| { !loaded() }
  class:translate-x-0=move|| { loaded() }
>
  hello
</h1>
```

**Comment by wongjn** [comment, marked as answer]
Yes, there is! You could set a property for `content.transform`, as per [the documentation](https://tailwindcss.com/docs/content-configuration#transforming-source-files):

> If you’re authoring content in a format that *compiles* to HTML (like Markdown), it often makes sense to compile that content to HTML *before* scanning it for class names.
>
> Use the `content.transform` option to transform any content matching a specific file extension before extracting classes:

Though, you could repurpose it slightly to work for the Leptos files. Tailwind has a default transformer as well as a built-in transformer for Svelte files. This is because Svelte has a similar syntax that would otherwise experience the same problem:

https://github.com/tailwindlabs/tailwindcss/blob/f1f419a9ecfcd00a2001ee96ab252739fca47564/src/lib/expandTailwindAtRules.js#L17

So you could do:

```js
export default {
  content: {
    transform: {
      rs: (content) => content.replace(/(?:^|\s)class:/g, ' '),
    },
    files: [
      // The `content` array would become `content.files` here.
    ],
  },
```

Otherwise, you could set a property for the `content.extract`, keyed by the file extension for the function that extracts the classes:

```js
export default {
  content: {
    extract: {
      rs: (context) => {
        // …
      },
    },
    files: [
      // The `content` array would become `content.files` here.
    ],
  },
```

For reference, here is the default extractor in Tailwind:

https://github.com/tailwindlabs/tailwindcss/blob/f1f419a9ecfcd00a2001ee96ab252739fca47564/src/lib/defaultExtractor.js#L3-L21

Though this method is undocumented and thus I'd assume not recommended.
